### PR TITLE
[Patient Intake Sample] Create `RelatedPerson` resource when necessary

### LINKED ISSUE #5161: [Patient Intake Sample] Logic limitation in patient intake flow
Current patient intake example supports spousal coverage for insurance - which is correct, as shown.  However, there is no opportunity to do data entry for the spouse - which will be required for correct billing.

image

---

Since we are creating a new `Patient`, we will have to create a new `RelatedPerson` as well:

1. Add a new subgroup to take the spouses information
2. Hide it using `Questionnaire.item.enableWhen`. It only needs to be shown if the subscriber is not the self
3. In the bot, parse these fields to create a new `RelatedPerson` resource

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Resolves #5161
- Create a `RelatedPerson` resource for the `Patient` depending on the `Coverage.relationship`

## Steps to test

- [ ] Build the bots: `npm run build:bots`
- [ ] Upload data
  image
- [ ] Set up patient coverage with a subscriber relationship (e.g., spouse, child, parent) to automatically create a `RelatedPerson` resource.

[Screencast from 02-09-2024 17:33:03.webm](https://github.com/user-attachments/assets/983a45db-75e7-40ca-be74-6ede02e87a6a)

### CODE REVIEW

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json [hunk 1]
```diff
@@ -257,8 +257,71 @@
                 "linkId": "relationship-to-subscriber",
                 "text": "Relationship to Subscriber",
                 "type": "choice",
-                "answerValueSet": "http://hl7.org/fhir/ValueSet/relatedperson-relationshiptype",
+                "answerValueSet": "http://hl7.org/fhir/ValueSet/subscriber-relationship",
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:260]: Drive-by fix based on [US Core Coverage Profile](https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-coverage.html)

image

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json [hunk 2]
```diff
@@ -257,8 +257,71 @@
...
                 "required": true
+              },
+              {
+                "linkId": "related-person",
+                "text": "Related Person Information",
+                "type": "group",
+                "enableBehavior": "all",
+                "enableWhen": [
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:268]: New group added to gather the related person information.

File: examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts [hunk 3]
```diff
@@ -605,3 +644,72 @@ export function convertDateToDateTime(date: string | undefined): string | undefi
...
+  }
+
+  // To simplify the demo, we're assuming the address is always a home address
+  return Object.keys(patientAddress).length > 0 ? { use: 'home', type: 'physical', ...patientAddress } : undefined;
+}
+
+function getRelatedPersonRelationshipFromCoverage(coverageRelationship: Coding): Coding | undefined {
+  // Basic relationship mapping between Coverage and RelatedPerson
+  // Coverage.relationship: http://hl7.org/fhir/ValueSet/subscriber-relationship
+  // RelatedPerson.relationship: http://hl7.org/fhir/ValueSet/relatedperson-relationshiptype
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts:699]: - https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-coverage.html
- https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-relatedperson.html

File: examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts:1]: Simple change: moved utils to `src/bots/core/intake-utils.ts`.

**Review by rahul1 — APPROVED** [review]
